lib/drip/client.rb

```diff
@@ -68,69 +68,31 @@ def content_type
       JSON_API_CONTENT_TYPE
     end
 
-    def get(url, options = {})
-      make_request(Net::HTTP::Get, make_uri(url), options)
-    end
-
-    def post(url, options = {})
-      make_request(Net::HTTP::Post, make_uri(url), options)
+    Drip::Request::VERB_CLASS_MAPPING.keys.each do |verb|
+      define_method(verb) do |path, options = {}|
+        warn "[DEPRECATED] Drip::Client##{verb} please use the API endpoint specific methods"
+        make_v2_request(verb, "v2/#{path}", options)
+      end
     end
 
-    def put(url, options = {})
-      make_request(Net::HTTP::Put, make_uri(url), options)
-    end
+  private
 
-    def delete(url, options = {})
-      make_request(Net::HTTP::Delete, make_uri(url), options)
+    def make_v2_request(http_verb, path, options = {})
```

Comment:
I think the name of the method doesn't reflects what it does. I'd suggest to have more generic `make_request` method that accepts different arguments, e.g:

```
def make_request(*args) ...
```

or

```
def make_request(request: nil, path: nil, verb: nil, options: nil) ...
```

Reply:
Heh, that was the original version before my changes…

I'm struggling with what to do with the content_type, which is effectively the only difference. I played with a couple of possibilities:
1. Pass the content type with every invocation of make_request. This felt very repetitive.
2. Create `JSON` and `JSONAPI` subclasses of `Drip::Request` that hard-code the content type. This felt cleaner, but I didn't like making subclasses to encode data.
3. Create a builder method for each type. This is what I ultimately landed on, as it felt like a variant of option 1 with some syntactic sugar to make it cleaner.

It sounds like you are suggesting that I go back to option 1. Do you think that is cleaner? Is there another way?

Reply:
I think `make_json_request` makes a lot of difference from `make_v2_request`, I'd like such name much more.

Reply:
I like that too. Change made!